Is AndroidManifest necessary and "ShowInstallPrompt"  can only  work after upload to google play?

Hi, I'm trying  to convert my existing Unity game into an Instant Game , and I Create an empty project by u3d 2018.2.5 , and create a loading scene to download assetbundle that contains my app's core game-play,
And I found that while I Change Android Build Type By 【PlayInstant/BuildSetting】and leave all blank default  then  Click Save, the Unity Console show the info "Saving Play Instant build settings..." and  had a warning message say "Assets/GooglePlayInstant/InstallLauncher.cs(82,13): warning CS0162: Unreachable code detected " , and  it did not  say message  like  "Successfully updated Assets/Plugins/Android/AndroidManifest.xml" ,And I wanna know Is the file AndroidManifest necessary In folder Plugins/Android ？ If its yes , What are the minimum requirements for file content?
Another thing confuse me is when I build The zip to upload to GooglePlay , I get apk and a mete-inf , 
and I install the apk in my device  and click a button that call the method "InstallLauncher.ShowInstallPrompt();" but nothing happened , its meats that this method can only work after upload to GooglePlay？
I'll appreciate it very much if you could help me！

--- a/GooglePlayInstant/Editor/BuildSettingsWindow.cs
+++ b/GooglePlayInstant/Editor/BuildSettingsWindow.cs
@@ -212,6 +212,8 @@ namespace GooglePlayInstant.Editor
             _assetBundleManifestPath = _assetBundleManifestPath.Trim();
             PlayInstantBuildConfiguration.SaveConfiguration(_instantUrl, scenesInBuild, _assetBundleManifestPath);
             PlayInstantBuildConfiguration.SetInstantBuildType();
+            Debug.Log("Saved Play Instant Build Settings.");
+
             // If a TextField is in focus, it won't update to reflect the Trim(). So reassign focus to controlID 0.
             GUIUtility.keyboardControl = 0;
             Repaint();
@@ -226,6 +228,7 @@ namespace GooglePlayInstant.Editor
         {
             PlayInstantBuildConfiguration.SetInstalledBuildType();
             _androidManifestUpdater.SwitchToInstalled();
+            Debug.Log("Switched to Android Build Type \"Installed\".");
         }
 
         private static void DisplayUrlError(string message)

--- a/GooglePlayInstant/Editor/PlayInstantBuildConfiguration.cs
+++ b/GooglePlayInstant/Editor/PlayInstantBuildConfiguration.cs
@@ -92,7 +92,6 @@ namespace GooglePlayInstant.Editor
         /// </summary>
         public static void SaveConfiguration(string instantUrl, string[] scenesInBuild, string assetBundleManifestPath)
         {
-            Debug.Log("Saving Play Instant build settings...");
             _config = _config ?? new Configuration();
             _config.instantUrl = instantUrl;
             _config.scenesInBuild = scenesInBuild;
@@ -182,6 +181,7 @@ namespace GooglePlayInstant.Editor
                 var oldInstantUrl = EditorPrefs.GetString(key);
                 SaveConfiguration(oldInstantUrl, null, null);
                 EditorPrefs.DeleteKey(key);
+                Debug.Log("Migrated Instant URL.");
             }
         }
     }

--- a/GooglePlayInstant/InstallLauncher.cs
+++ b/GooglePlayInstant/InstallLauncher.cs
@@ -52,7 +52,7 @@ namespace GooglePlayInstant
         /// </summary>
         public static void ShowInstallPrompt(string referrer = null)
         {
-            using (var activity = GetCurrentActivity())
+            using (var activity = UnityPlayerHelper.GetCurrentActivity())
             using (var postInstallIntent = CreatePostInstallIntent(activity))
             {
                 ShowInstallPrompt(activity, IgnoredRequestCode, postInstallIntent, referrer);
@@ -76,10 +76,7 @@ namespace GooglePlayInstant
         public static void ShowInstallPrompt(
             AndroidJavaObject activity, int requestCode, AndroidJavaObject postInstallIntent, string referrer)
         {
-#if !PLAY_INSTANT
-            return;
-#endif
-
+#if PLAY_INSTANT
             if (activity == null)
             {
                 throw new ArgumentNullException("activity");
@@ -110,6 +107,9 @@ namespace GooglePlayInstant
                         requestCode);
                 }
             }
+#else
+            throw new InvalidOperationException("ShowInstallPrompt should only be called from instant apps.");
+#endif
         }
 
         [Obsolete("Use UnityPlayerHelper.GetCurrentActivity() instead.")]

--- a/GooglePlayInstant/Samples/SphereBlast/scripts/BaseGame.cs
+++ b/GooglePlayInstant/Samples/SphereBlast/scripts/BaseGame.cs
@@ -13,11 +13,9 @@
 // limitations under the License.
 
 using System;
-using System.Collections;
-using System.Collections.Generic;
-using System.Linq.Expressions;
 using GooglePlayInstant;
 using UnityEngine;
+using UnityEngine.SceneManagement;
 using UnityEngine.UI;
 using Random = UnityEngine.Random;
 
@@ -130,7 +128,7 @@ public class BaseGame : MonoBehaviour
 
     private void UpdateTimer()
     {
-        _timeLeftText.text = "" + System.Math.Max(_timer, 0);
+        _timeLeftText.text = Math.Max(_timer, 0).ToString();
     }
 
     /* SCENE SET UP METHODS */
@@ -144,18 +142,15 @@ public class BaseGame : MonoBehaviour
 
     private static void WriteGameStateToCookie(int level, int score)
     {
-#if UNITY_EDITOR
-        return;
-#endif
+#if !UNITY_EDITOR
         var gameStateCsv = level + "," + score;
         CookieApi.SetInstantAppCookie(gameStateCsv);
+#endif
     }
 
     private void ReadGameStateFromCookie()
     {
-#if UNITY_EDITOR
-        return;
-#endif
+#if !UNITY_EDITOR
         var results = CookieApi.GetInstantAppCookie();
 
 
@@ -173,6 +168,7 @@ public class BaseGame : MonoBehaviour
         _score = int.Parse(cookieScore);
         Debug.Log("CookieLevel: " + cookieLevel);
         Debug.Log("CookieScore: " + cookieScore);
+#endif
     }
 
     /* LEVEL METHODS */
@@ -187,7 +183,7 @@ public class BaseGame : MonoBehaviour
         _level = newLevel;
         _maxTime = _levelTimeLimitMapping[_level - 1];
         _timer = _maxTime;
-        _levelText.text = "" + _level;
+        _levelText.text = _level.ToString();
     }
 
     public int GetLevel()
@@ -210,7 +206,7 @@ public class BaseGame : MonoBehaviour
     public void UpdateScoreDisplay()
     {
         var displayScore = _score % MAX_SCORE_PER_LEVEL;
-        _scoreText.text = "" + displayScore;
+        _scoreText.text = displayScore.ToString();
     }
 
     private void SetScore(int newScore)
@@ -236,7 +232,7 @@ public class BaseGame : MonoBehaviour
 
     public void RestartGame()
     {
-        Application.LoadLevel("SphereScene");
+        SceneManager.LoadScene("SphereScene");
     }
 
     public void IncrementScore()

